File: module3/VisMealy.java
import java.util.Scanner;

public class VisMealy {
    public static void outPut(String[][] exit, int[][] crossover, int start){
        System.out.println("digraph {");
        System.out.println("    rankdir = LR");
        for (int i = 0; i < crossover.length; i++) {
            for (int j = 0; j < crossover[i].length; j++) {
                if (!(exit[i][j]==null)){
                    System.out.print("    "+i + " -> " + +crossover[i][j] + "[label = \"" + (char) (j + 97) + "(" + (exit[i][j]) + ")" + "\"]\n");
                }
            }
        }
        System.out.println("}");
    }
    public static void main(String[] args) {
        int n,m,start,cr;
        String sym;
        Scanner scanner = new Scanner(System.in);

        n = scanner.nextInt();
        m = scanner.nextInt();
        start = scanner.nextInt();
        int[][] crossover = new int[2*n][m];
        String[][] exit = new String[2*n][m];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                crossover[i][j] =scanner.nextInt();
            }
        }
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                exit[i][j] =scanner.next();

            }
        }
        outPut(exit,crossover,start);
    }

}

File: module3/VisMealy.go
package main

import "fmt"
func ratatatata(exit [][]string, crossover [][]int, start int)  {
	fmt.Println("digraph {")
	fmt.Println("rankdir = LR")


	
	for i, _ := range crossover {
		for j, x := range crossover[i] {
			fmt.Printf("%d -> %d [label = \"%s(%s)\"]\n", i, x, string(j + 97), exit[i][j])
		}
	}
	fmt.Println("}")
}
func main() {
	var n, m, start, cr int
	var sym string
	_, _ = fmt.Scan(&n)
	crossover := make([][]int, n)
	exit := make([][]string, n)
	_, _ = fmt.Scan(&m)
	_, _ = fmt.Scan(&start)
	for i := 0; i < n; i++ {
		crossover[i] = make([]int, 0)
		for j := 0; j < m; j++ {
			_, _ = fmt.Scan(&cr)
			crossover[i] = append(crossover[i], cr)
		}
 	}
	for i := 0; i < n; i++ {
		exit[i] = make([]string, 0)
		for j := 0; j < m; j++ {
			_, _ = fmt.Scan(&sym)
			exit[i] = append(exit[i], sym)
		}
	}
	ratatatata(exit, crossover, start)
 }
